Add argument assertions to inspect/status/url commands

kechol opened the pull request:
Add assertions for no-argument case on `inspect`, `status` and `url` commands to fix unkind messages.

Current message is:
```
$ docker-machine status
unable to load host: open /Users/user/.docker/machine/machines/config.json: no such file or directory
```

To be fixed like:
```
$ docker-machine status
Error: Expected one machine name as an argument.
```

-----
os: OSX Yosemite 10.10.5
docker-machine version: 0.4.1 (e2c88d6)

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "add-argument-assertion" git@github.com:kechol/machine.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

nathanleclaire commented:
Looks pretty good to me.  I'd rather have it than not and clean it up later if we introduce more universal parsing for this sort of thing.

WDYT @ehazlett ?

nathanleclaire commented:
@kechol Any chance I could get you to add some integration tests for this to the `test/integration/cli` folder?  Other than that, LGTM

kechol commented:
@nathanleclaire 
Added some integration tests, sorry for my late reply. There is already common cases in test/integration/core folder so I just added simple error cases for this, thanks.

nathanleclaire commented:
LGTM

Thanks @kechol !
